src/openrct2/actions/ScenarioSetSettingAction.hpp

```diff
@@ -0,0 +1,290 @@
+/*****************************************************************************
+ * Copyright (c) 2014-2019 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#pragma once
+
+#include "../OpenRCT2.h"
+#include "../interface/Window.h"
+#include "../management/Finance.h"
+#include "../peep/Peep.h"
+#include "../world/Park.h"
+#include "GameAction.h"
+
+#include <algorithm>
+
+enum class ScenarioSetSetting : uint8_t
+{
+    NoMoney,
+    InitialCash,
+    InitialLoan,
+    MaximumLoanSize,
+    AnnualInterestRate,
+    ForbidMarketingCampaigns,
+    AverageCashPerGuest,
+    GuestInitialHappiness,
+    GuestInitialHunger,
+    GuestInitialThirst,
+    GuestsPreferLessIntenseRides,
+    GuestsPreferMoreIntenseRides,
+    CostToBuyLand,
+    CostToBuyConstructionRights,
+    ParkChargeMethod,
+    ParkChargeEntryFee,
+    ForbidTreeRemoval,
+    ForbidLandscapeChanges,
+    ForbidHighConstruction,
+    ParkRatingHigherDifficultyLevel,
+    GuestGenerationHigherDifficultyLevel,
+    Count
```

Comment:
For an `enum class`, `Count` should not be part of the `enum`. The compiler will complain if you don't handle it as a case, etc.

Reply:
This is how we are doing it for OpenRCT2. We need to be able to know the size of the enum.

Reply:
Only those which aren't `enum class`. I suppose the idea is that `enum class` represents a value of which should never no actual meaningful underlying value. I have used `constexpr size_t` underneath to this in other cases.

Reply:
All of the other game actions are using this method. I don't especially like it either but was going for consistency.

Reply:
What do you need a count for?

Reply:
Since you are only using `count` to check if it is valid I would instead just return the invalid result using the default case.

Reply:
I'd rather keep it matching the rest of the actions for now and then refactor them all in a oner later.

Reply:
I think its perfectly fine to do this, switches should have a default block where you should report some error because its an unexpected case, also defaulting the value to Count will also provide a way to tell if the variable was never assigned.